### ISSUE TITLE
concealcursor not set from ftplugin when loaded via bufadd+bufload+nvim_win_set_buf

### ISSUE THREAD

**Issue by taylan97** [neovim/neovim#13201]
### Bug

Setting a custom `concealcursor` value in an after/ftplugin does not "stick" when a buffer is loaded via bufadd + bufload. Other values from the ftplugin _are_ set correctly.

- `nvim --version`: v0.5.0-773-g932585fd9
- Operating system/version: WSL/Ubuntu
- Terminal name/version: alacritty
- `$TERM`: xterm-256color

### Steps to reproduce using `nvim -u NORC`

See https://github.com/taylan97/concealcursor-bug.vim for reproduction fixture and instructions.

_This may not be a bug, it's unclear if loading a file via `bufadd -> bufload -> nvim_win_set_buf` is intended to load ftplugins or not. It definitely does apply some settings from the ftplugins file, so probably._

### Actual behaviour

`concealcursor `value is not correctly set from `after/ftplugin` file.

### Expected behaviour

`concealcursor `value should be set according to ftplugin.

Thank you for your time.

**Comment by bfredl** [comment]
same root issue as #10070 ?

**Comment by taylan97** [comment]
Yes perhaps. Though your inspections in that thread talk about `nvim_win_set_buf`'s behavior with *unloaded* buffers, while this buffer is loaded (via `bufload` at least, which to me (and the docs) certainly *implies* its being loaded).

What's strangest about this issue is that some of the ftplugin settings *are* run (i.e. `showbreak`). So the file is obviously being executed, just perhaps in some sandbox/isolated state. 

I have tested it with both `set` and `setlocal` to the same result.

If I add `autocmd BufEnter * echo "message"` to `after/ftplugin/mardown.vim`, the message is echo'd out no matter the method of loading ("b" or "n").

**Comment by taylan97** [comment]
Also serendipitous,  #12951 "Files in after/ftplugin not loaded". I wonder if it's an issue with `after/`* specifically. 

In that issue, you are instructed to setup the ftplugin, then run `:copen` (not `:e <file>`!), which might be performing a similar "non standard" load?

**Comment by taylan97** [comment]
Tried re-configuring the bug fixture.

It presents identically, whether it's ftplugin is in `$RTP/ftplugin` or `$RTP/after/ftplugin`, so I don't think this is specific to `after` style plugins.

**Comment by bfredl** [comment]
@taylan97 you are right, it is not exactly the same issue. But ideally we can refactor the code directly to fix both issues at once.

**Comment by justinmk** [comment]
Steps to reproduce this are complicated. @taylan97 can you check if this still is present on the development release? 

- The [Releases](https://github.com/neovim/neovim/releases/tag/nightly) page has pre-built archives for Linux/Windows/macOS. 
- Check `:version` to confirm that you are using the latest version.
